zio-http/src/main/scala/zhttp/http/HttpResult.scala

```diff
@@ -94,11 +201,13 @@ object HttpResult {
 
   // EVAL
   @tailrec
-  private[zhttp] def evaluate[R, E, A](result: HttpResult[R, E, A]): Out[R, E, A] = {
+  private[zhttp] def evaluate[R, E, A](
+    result: HttpResult[R, E, A],
+  )(implicit ev: NeedsEnv[R]): Out[R, E, A] = {
     result match {
-      case m: Out[_, _, _]         => m
-      case Suspend(r)              => evaluate(r())
-      case FoldM(self, ee, aa, dd) =>
+      case m: Out[_, _, _]                       => m
+      case Suspend(r)                            => evaluate(r())
```

Comment:
Similar to `Suspend` you can add a case for `Provide(a, r)`, it should look something like —

```scala
  case Provide(a, r) => 
    a match {
      case a: Out[_, _, _] => a
      case Effect(z)       => evaluate(z.provide(r))
      case FoldM(self, ee, aa, dd) => self.foldM( ... )
    }
```